Css update and round helper buttons

chromoxdor opened the pull request:
Ok, last time i try to make an css related pull request... 🙂
The css is roughly the same size than the old one but has a lot of advantages.
I also added the round helper buttons to this branch.
<img width="795" alt="css" src="https://user-images.githubusercontent.com/33860956/184345559-2df76ab1-0096-4e28-b060-8fd391df9291.png">

chromoxdor committed:
changed css +round buttons

chromoxdor committed:
css update+ round helper buttons

TD-er commented:
Hmm there is quite a lot of changes in the CSS, which wasn't clear from the screenshot.

I have to say that it does look more clean, even though it is a bit to get to used to.

The docs buttons are a bit strangely positioned here:
![image](https://user-images.githubusercontent.com/3751318/184351334-c1f47ebc-27e1-45cc-8c82-4c67137fc3f3.png)


Ditto:
![image](https://user-images.githubusercontent.com/3751318/184351475-ae305726-88be-436a-b78a-f55271a51ee2.png)

TD-er commented:
This also does seem a bit strange now with rounded corners:

![image](https://user-images.githubusercontent.com/3751318/184351820-8f88cb2e-0f47-4e2a-bceb-bcb41b3ea193.png)

chromoxdor committed:
removed unnecessary css

TD-er commented:
I find the highlight color a bit too strong.
Take for example this from the timing stats page (also the root page showing the other nodes)
![image](https://user-images.githubusercontent.com/3751318/184354845-287a931d-bdd2-458b-bebc-2b0ec3050579.png)

The contrast between background and font is a bit too low.
That's why I originally picked the lighter green-ish tint to make it better readable.

Also it seems there is no transparency, so the slight difference in alternating rows is lost.

Just tweaked a bit in the inspector tab and came up with `#ff8d2bcc`
Just as a suggestion.

TD-er commented:
Another one, which has become more clear....

```c++
void devicePage_show_task_values(taskIndex_t taskIndex, deviceIndex_t DeviceIndex)
{
  // section: Values
  const uint8_t valueCount = getValueCountForTask(taskIndex);

  if (!Device[DeviceIndex].Custom && (valueCount > 0))
  {
    int colCount = 1;
    addFormSubHeader(F("Values"));
    html_end_table();
    html_table_class_normal();
```

Can you change the init of `colCount` to 2? 
```c++
    int colCount = 2;
```

The line at the bottom of the task values is 1 column too short.
I can fix it too, but it fits nicely here in this PR.

chromoxdor commented:
```
The line at the bottom of the task values is 1 column too short.
I can fix it too, but it fits nicely here in this PR.
```

on it...

`The contrast between background and font is a bit too low.`
ubbs...actually i did.. but it seems i edited the wrong file 😳

chromoxdor committed:
changed highlightcolor  + wider column

chromoxdor commented:
so..changed the color & made the column line wider
i choose a  more.... "friendly" looking highlight color :)
What do you think?
<img width="893" alt="color" src="https://user-images.githubusercontent.com/33860956/184361033-28b43479-e6ac-4be9-9883-aed386ae9e2a.png">

TD-er commented:
I get why you may want to do it like this on mobile devices
![image](https://user-images.githubusercontent.com/3751318/184361161-2a6c6db5-c858-400c-87e3-61cc0903c1cd.png)

But on a desktop, these are way too wide.

chromoxdor commented:
`But on a desktop, these are way too wide.`

shit.. i changed that last minute because i wanted to accommodate  the "xwide"commit.

In my original css it looks like this and has nothin to do with the mobile adjustments...

<img width="891" alt="Bildschirmfoto 2022-08-12 um 15 17 06" src="https://user-images.githubusercontent.com/33860956/184361852-82f17070-6b44-422c-b37b-dd9db1048bb7.png">

chromoxdor committed:
fixed to wide input fields

chromoxdor commented:
i reverted that change...

chromoxdor commented:
I should concentrate on only one project today.. together with the heat i easily get confused  🤪
Remember the ESPeasy-powered battery cabinet...lots of cabling  😳
![IMG_1502](https://user-images.githubusercontent.com/33860956/184363076-decdfe02-aa36-4492-87ee-edf62c88ca26.JPG)

tonhuisman commented:
> does seem a bit strange now with rounded corners

@chromoxdor did you recently switch to Windows 11 or MacOS inspired? 🤣

tonhuisman commented:
> ..lots of cabling

And lots of hot-glue, on a hot day 🤣

chromoxdor commented:
> > does seem a bit strange now with rounded corners
> 
> @chromoxdor did you recently switch to Windows 11 or MacOS inspired? 🤣

Give me some break guys...the input fields and buttons have rounded corners so i followed that design...
But i can definetly revert that. You are a tough audience....
😁

tonhuisman commented:
> What do you think?

Can we have the original greenish-yellow color back? Orange feels like a warning color. The current color just is nice enough, *and* looks more familiar...

tonhuisman commented:
> You are a tough audience....
> 😁

Yes 🤔 😛

chromoxdor commented:
> And lots of hot-glue, on a hot day 🤣

With a really shitty hotglue gun 😩 ..... 

Soldering the rows of illuminated buttons was also a challenge...
![IMG_1510](https://user-images.githubusercontent.com/33860956/184365286-f997d046-fd4e-4320-b9f1-91b6f5327223.JPG)

chromoxdor commented:
`Can we have the original greenish-yellow color back? `
of course you can! the main thing is that something has finally been moved in terms of css

tonhuisman commented:
To fix the `Hardware Settings` buttons issue, you'll have to adjust this:

![image](https://user-images.githubusercontent.com/1922393/184367430-b2b1da5d-da56-4b34-8e7a-b96dd134801d.png)

from 225 to something like 300 in file `src/src/WebServer/Markup.cpp`

chromoxdor committed:
no rounded corners and green is back

no rounded corners for the gentlemen programmers

chromoxdor commented:
Here you go:
<img width="894" alt="Bildschirmfoto 2022-08-12 um 15 55 14" src="https://user-images.githubusercontent.com/33860956/184368319-e6ea55c4-cfe7-477d-ab16-4f65f68dad8d.png">

chromoxdor commented:
> To fix the `Hardware Settings` buttons issue, you'll have to adjust this:

On second earlier.....but i decided to leave the cabinet so the change is on its way

chromoxdor committed:
Update Markup.cpp

a bit wider to have everything in one line

chromoxdor commented:
`@chromoxdor did you recently switch to Windows 11 or MacOS inspired? 🤣`

😁
<img width="892" alt="Bildschirmfoto 2022-08-12 um 16 15 43" src="https://user-images.githubusercontent.com/33860956/184372520-be381b6d-895c-44e1-a4fd-d9c52106cb44.png">

tonhuisman commented:
If the browser width is between 750 and 785 px (I'm on 100% display setting, on a Full HD 15.6" Windoze laptop) I get this overflow of the Tools tab:
![image](https://user-images.githubusercontent.com/1922393/184372857-ab29f946-1b6f-4eb7-b5f9-9926f97dd437.png)

might need some fine-tuning?

When narrower than 750px the media query kicks in and the text labels are suppressed, then it's OK again.

chromoxdor commented:
> If the browser width is between 750 and 785 px (I'm on 100% display setting, on a Full HD 15.6" Windoze laptop) I get this overflow of the Tools tab:

Interesting.. i can’t  reproduce this but i would guess that this is because the icons are rendered differently on other systems...i´ll give it a bit more safety margin...

tonhuisman commented:
> `@chromoxdor did you recently switch to Windows 11 or MacOS inspired? 🤣`
> 
> 😁 <img alt="Bildschirmfoto 2022-08-12 um 16 15 43" width="892" src="https://user-images.githubusercontent.com/33860956/184372520-be381b6d-895c-44e1-a4fd-d9c52106cb44.png">

That just indicates you should (also) update PR #4088 😉

chromoxdor committed:
accommodate to windows users

chromoxdor commented:
`That just indicates you should (also) update PR https://github.com/letscontrolit/ESPEasy/pull/4088 😉`
i can´t... i use it for a longe time now for myself but as i stated there, i do not know how to implement it the right way.
Right now it is directing directly to a cdn from the webserver.cpp...but actually it should if static files available direct to them. instead to a cdn. 
There i need definitely help... 

That’s how i implemented it for myself:
<img width="1165" alt="Bildschirmfoto 2022-08-12 um 16 39 17" src="https://user-images.githubusercontent.com/33860956/184379317-4c8229b8-d64f-4906-bab3-f3d6f35bc8d4.png">

tonhuisman commented:
Back on those rounded corners (☺️), I think most of *my* 'resistance' there came from the rather wide (5px) padding around it. That doesn't feel mobile-friendly, taking quite some width-space. While that padding IMHO is fine for bottom and top, the left and right padding is a bit much, a 0 or 1 padding might just be enough there.

chromoxdor commented:
🙂
<img width="829" alt="Bildschirmfoto 2022-08-12 um 16 43 47" src="https://user-images.githubusercontent.com/33860956/184379801-21182769-b9b4-4a8c-b238-630001f7e978.png">

chromoxdor commented:
> While that padding IMHO is fine for bottom and top, the left and right padding is a bit much, a 0 or 1 padding might just be enough there.

The Problem with a missing padding on the sides is, that scrolling down the rulespage to the save button becomes a bit harder...

chromoxdor committed:
put back some stuff

i striped my css down a lot to make it the same size as the old one.... but for better user experience i put some things back in :)

chromoxdor committed:
ubbs..i edited the wrong line

TD-er commented:
I ran esp.css through [this beautifier](https://www.cleancss.com/css-beautify/)
It also made quite a few suggestions like "this combination may make objects larger than expected" and some compatibility issues. 
Even apparently a typo:
![image](https://user-images.githubusercontent.com/3751318/184432431-c7ad7eab-225d-4639-8e0e-5e52aeb026bc.png)


Maybe you can have a look at those too?

TD-er commented:
This [online CSS optimizer](https://www.cssportal.com/css-optimize/) also makes some suggestions and does reduce the raw CSS quite a bit already. Not sure if minify afterwards makes it much smaller, but it could be worth trying.

Edit:
Running the optimizer + minify makes the CSS about 160 bytes smaller.

chromoxdor commented:
Interesting.. i´ll have a look at it.

chromoxdor committed:
modernized the css

chromoxdor commented:
@TD-er  Thanks for the links.. we saved 544byte :)

chromoxdor commented:
> 🙂 <img alt="Bildschirmfoto 2022-08-12 um 16 43 47" width="829" src="https://user-images.githubusercontent.com/33860956/184379801-21182769-b9b4-4a8c-b238-630001f7e978.png">

By the way, i forgot to address you directly ..sorry 
If you have time, can you help to implement https://github.com/letscontrolit/ESPEasy/pull/4088

tonhuisman reviewed (approved):
A lot of small, subtle improvements. It's looking very good 💯

chromoxdor commented:
> A lot of small, subtle improvements. It's looking very good 💯

Thank you... 
for 6.285kb i could throw in a auto-dark mode but i guess with the chronic memory shortage this is not a good idea. 

Edit: right now we are at 4.8kb
<img width="837" alt="Bildschirmfoto 2022-08-13 um 12 16 33" src="https://user-images.githubusercontent.com/33860956/184479441-c9c723a4-c1d9-4821-8532-04e2401ae7c4.png">

tonhuisman commented:
You can still upload a `esp.css` file on the filesystem, that will be loaded automatically.

Including clear&simple instructions on how to do that will help most that want to use (auto-)dark mode. 🕶️ Retrieving that file is probably the most difficult part, but if you name it properly, and put it in the `static` folder it shouldn't be too hard. By *not* including it in the `WebStaticData.h` file, you won't have that extra 4.8kB bothering you.

chromoxdor committed:
render table-border on ios devices

TD-er commented:
> > A lot of small, subtle improvements. It's looking very good 💯
> 
> Thank you... for 6.285kb i could throw in a auto-dark mode but i guess with the chronic memory shortage this is not a good idea.
> 
> Edit: right now we are at 4.8kb <img alt="Bildschirmfoto 2022-08-13 um 12 16 33" width="837" src="https://user-images.githubusercontent.com/33860956/184479441-c9c723a4-c1d9-4821-8532-04e2401ae7c4.png">

About this dark theme...
Does it differ a lot?
For example we can add something to the class names in the HTML to mark dark mode?
Like add some wrapper class surrounding everything, which may then set some base colors you can then refer to in the CSS?

Maybe something like [using var() function in CSS](https://stackoverflow.com/questions/1875852/how-can-i-define-colors-as-variables-in-css)

chromoxdor committed:
table-border ios fix the 2nd

chromoxdor commented:
> Does it differ a lot?

not at all.. its the same file just added the different colors through "@media (prefers-color-scheme: dark)"
I think it’s  the easiest option to do it!? It doesn’t get smaller any other way (i think).
so roughtly 1,5k more...

chromoxdor commented:
we have to change a lot of colors to make it look nice...

edit: i could remove some maybe to find a compromise between nice and small

TD-er commented:
Have you seen the link I posted to some stackoverflow question regarding using variables in CSS?

chromoxdor commented:
oh… not really.  i’ll look into it…. tomorrow :)

chromoxdor commented:
@TD-er In general i find the idea to have switch and using css vars very good.
In preparation for that i optimized the code further. I think it doesn’t get any smaller.
The original default.css: 5.2k
The css of this branch: 4.8k
The new css with dark mode via media-query: 5.8k (With variables it wouldn’t be smaller)


In order to have a switch somebody (not me because unfortunately i am unable to do so) would need to implement it in the firmware. 
If a dark mode is wanted (personally i don’t really care since i do customize builds with my css anyway) the easiest approach would be via media query since i already have the css for that and no extra code is needed.
But i would/could also provide a css with variables if you or anyone else implements it...

chromoxdor committed:
Merge branch 'mega' into css-update-and-round-helper-buttons

chromoxdor committed:
Merge branch 'letscontrolit:mega' into css-update-and-round-helper-buttons

TD-er committed:
Merge branch 'mega' into css-update-and-round-helper-buttons

TD-er committed:
[ESP32 MAX] Fix not defining LIMIT_BUILD_SIZE on MAX builds

TD-er commented:
@chromoxdor Congrats with your first merged PR, let there be many more ;)

chromoxdor commented:
> @chromoxdor Congrats with your first merged PR, let there be many more ;)

Hey @TD-er , today is the last day of my vacation and i am reading this right now. Wonderful! 
It makes me a bit proud and i thank you for letting me contributing to this project.
